### PR TITLE
Add payment requests aka invoices

### PR DESCRIPTION
**Pull request by zefir-git**
# Commands
## `/invoice help`
Show invoice commands you have permission to use

## `/invoice create <account> <amount> [description] [--player <player>]`
Create an invoice. With `--player <player>` you can set who can pay the invoice (and they get a notification). If you don't set this, anyone who knows the invoice ID can pay it. With `<account>` you select the account where you want to receive the money.

### Permissions:
- `bank.invoice.create` - create an invoice
- `bank.invoice.create.other` - allows creating an invoice with a recipient account that you don't own

## `/invoice view <invoice>`
See details of an invoice.

### Permissions:
- `bank.invoice.view` - view invoices in which you are the seller or the buyer
- `bank.invoice.view.other` - view any invoice

## `/invoice pay <invoice> <account>`
Pay an invoice using an account of your choice

### Permissions:
- `bank.transfer.self` - pay an invoice that is issued from one of your other accounts
- `bank.transfer.other` - pay an invoice issued by a 3rd party
- `bank.invoice.pay.other` - pay an invoice sent to a player that is not you
- `bank.invoice.pay.account-other` - pay an invoice using somebody else's account

## `/invoice send <invoice> <player>`
Send an existing invoice (in which you are the seller) to a player. This just sends a message and can be used for a reminder.

If the invoice specifies a buyer, you can only send it to the same buyer or to players that have the `bank.invoice.view.other` permission.

### Permissions:
- `bank.invoice.send` - allows sending an invoice
- `bank.invoice.send.other` - allows sending an invoice that you did not create

## `/invoice list [all|sent|received] [page] [--player <player>]`
List invoices. With `all` (the default) you see both invoices you've created and are expected to pay/have paid. With `sent` you see just the former and with `received` only the latter. The page argument defaults to 1. With `--player` you can see all/sent/received invoices of another player (if you have the permission).

### Permissions:
- `bank.invoice.view` - list your invoices
- `bank.invoice.view.other"` - list invoices of another player

### CODE REVIEW

**Review by km127pl — DISMISSED** [review]
Looks good to me

**Review by km127pl — APPROVED** [review]
lgtm

File: src/main/resources/config.yml [hunk 1]
```diff
@@ -488,6 +506,82 @@ messages:
...
+            <white>From: <hover:show_text:'<white><account></white><newline><gray><account-type></gray><newline><gray>Owned by <account-owner></gray><newline><gray><account-id></gray>'><gray><account></gray></hover>
+            <white>To: <gray><buyer></gray></white>
+            <white>Date: <gray><date:'yyyy-MM-dd HH:mm'> UTC</gray></white>
+            <white>Status:</white> <status>
+            <white>Description: <gray><description></gray></white><pay-button>
+
+        # Used in the <status> placeholder
+        # Same placeholders as details (except <status>)
+        status:
+            paid: '<green>Paid</green> <gray>(transaction #<transaction-id>)</gray>'
```
Comment (Dviih) [inline src/main/resources/config.yml:542]:
```suggestion
            paid: "<green>Paid</green> <gray>(transaction #<transaction-id>)</gray>"
```
Double quotes are nicer

**Review by Dviih — DISMISSED** [review]
It all seems good to me.

**Review by Dviih — APPROVED** [review]
Seems good to me

**Pull request merged**